### PR TITLE
IDEX-1620: Disable buttons when got branch list

### PR DESCRIPTION
**Pull request by vinokurig**

### CODE REVIEW

**Comment by skabashnyuk** [comment]
LGTM

**Comment by sleshchenko** [comment]
Not OK for me. These changes produce set of bugs. For example, when user click on 'Create Branch' button and create new branch then after closing 'Create new branch' window all branches is disabled. You can use mechanism like https://github.com/codenvy/che-plugins/blob/IDEX-1620/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/remote/RemoteViewImpl.java#L218

So, view will check actuality of selection on setBranches. If selection is not actual view will send notice to presenter about it. Then presenter will disable buttons.

**Comment by vinokurig** [comment]
fixed by adding a check that branch is not selected

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java [hunk 1]
```diff
@@ -309,6 +308,13 @@ protected void onFailure(Throwable exception) {
         }, null).show();
     }
 
+    @Override
+    public void onBranchUnselected() {
```
Comment (sleshchenko) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java]: I think you should reset value of current branch in this method.

**Comment by sleshchenko** [comment]
other OK for me

**Comment by skabashnyuk** [comment]
LGTM what about @vparfonov

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java [hunk 2]
```diff
@@ -104,11 +104,7 @@ public BranchPresenter(BranchView view,
     /** Show dialog. */
     public void showDialog() {
```
Comment (vparfonov) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchPresenter.java]: maybe need change method name, now it don't show dialog

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchView.java [hunk 3]
```diff
@@ -83,6 +86,14 @@
     /** Close dialog. */
     void close();
 
+    /**
```
Comment (vparfonov) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/branch/BranchView.java]: Do we really need this method like public?

**Comment by vparfonov** [comment]
ok

**Pull request merged**